### DISCUSSION TITLE
How to find device address by device name, on Mac OSX?

### DISCUSSION THREAD

**Discussion by zapta** [hbldh/bleak#1247] [Q&A]
My app needs to connect to devices given their names. On Windows and Linux, the address is constructed from the name, but this doesn't work on Max OSX where the OS uses arbitrary device addresses.  As a workaround, I am using the code below which scan for 5 seconds and then look the address by name.

My questions are:

1. Is this approach reasonable or is there a simpler approach? E.g. a connect by name API function.

3. Currently the scanning takes 5 secs even if the device was detected after 1 sec. Is there a way to abort the scan once the device is detected?

```
    devices = await BleakScanner.discover(timeout=5)
    for device in devices:
        if device.name == value:
            return device.address
```

**Comment by dlech** [comment]
https://bleak.readthedocs.io/en/latest/api/scanner.html#bleak.BleakScanner.find_device_by_name

**Comment by zapta** [comment]
Thanks @dlech .

**Comment by zapta** [comment]
@dlech, was it released?  

I can't find find_device_by_name in bleak 0.19.5.

**Reply by dlech** [reply]
https://github.com/hbldh/bleak/blob/develop/CHANGELOG.rst

Looks like it hasn't been released yet.

**Comment by zapta** [comment]
NP. I copied the implementation to my code and it works. Thanks.
